src/app/features/courses/courses.component.ts

```diff
@@ -1,47 +1,67 @@
-import { Component, Input, OnInit } from "@angular/core";
-import { ButtonTypes } from "@app/shared/types/button.type";
-import { CourseInfo } from "@app/shared/types/course.model";
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { CoursesStoreService } from '@app/services/courses-store.service';
+import { MappingService } from '@app/services/mapping.service';
+import { Author } from '@app/shared/types/author.model';
+import { ButtonTypes } from '@app/shared/types/button.type';
+import { Course, CourseInfo } from '@app/shared/types/course.model';
+import { UserStoreService } from '@app/user/services/user-store.service';
+import { combineLatest, Subscription } from 'rxjs';
 
 @Component({
-  selector: "app-courses",
-  templateUrl: "./courses.component.html",
-  styleUrls: ["./courses.component.scss"],
+  selector: 'app-courses',
+  templateUrl: './courses.component.html',
+  styleUrls: ['./courses.component.scss'],
 })
-export class CoursesComponent implements OnInit {
+export class CoursesComponent implements OnInit, OnDestroy {
   ButtonTypes = ButtonTypes;
+  private subscription: Subscription | undefined;
 
-  @Input() courses: CourseInfo[] = [];
+  allCourses: Course[] = [];
+  allAuthors: Author[] = [];
+  courses: CourseInfo[] = [];
+  isAdmin$ = this.userStore.isAdmin$;
+  isLoading$ = this.coursesStore.isLoading$;
 
   courseInfo: CourseInfo | undefined = undefined;
   filteredCourses: CourseInfo[] = [];
-  lastSearchedText = "";
+  lastSearchedText = '';
   notFound = false;
```

Comment:
Maybe this becomes obsolete once you get the response from API for search

Reply:
I kept this , because for no course result I want to show rather all courses and informing user there was no match...
As filter result must be reflected on the course service's courses observable , if I get no result and set courses to an empty array it would lead to the content of 'No courses content suggesting to add new courses' which would be a misleading information...